lib/PreviewProvider.js

```diff
@@ -211,9 +169,10 @@ PreviewProvider.prototype = {
       this._asyncSaveLinks(processedLinks, event);
     }
 
-    let cachedLinks = this._getEnhancedLinks(processedLinks, previewsOnly, event);
-    return this._getFaviconColors(cachedLinks).then(linksToSend => {
-      return linksToSend;
+    return this._asyncGetEnhancedLinks(processedLinks, previewsOnly, event).then(cachedLinks => {
+      return this._getFaviconColors(cachedLinks).then(linksToSend => {
+        return linksToSend;
```

Comment:
No I mean that .then(stuff => stuff) is a no-op, you can just remove that.  I just tried and the tests still pass.